vms/registration/tests/test_functional_volunteer.py

```diff
@@ -272,56 +312,98 @@ def test_city_field(self):
+
+    def test_phone_field(self):
+
+        # register valid volunteer user with valid phone number for country
+        self.driver.get(self.live_server_url + self.volunteer_registration_page)
 
-        # test special characters in city
+        self.driver.find_element_by_id('id_username').send_keys('volunteer-username')
+        self.driver.find_element_by_id('id_password').send_keys('volunteer-password!@#$%^&*()_')
+        self.driver.find_element_by_id('id_first_name').send_keys('volunteer-first-name')
+        self.driver.find_element_by_id('id_last_name').send_keys('volunteer-last-name')
+        self.driver.find_element_by_id('id_email').send_keys('volunteer-email@systers.org')
+        self.driver.find_element_by_id('id_address').send_keys('volunteer-address')
+        self.driver.find_element_by_id('id_city').send_keys('volunteer-city')
+        self.driver.find_element_by_id('id_state').send_keys('volunteer-state')
+        self.driver.find_element_by_id('id_country').send_keys('India')
+        self.driver.find_element_by_id('id_phone_number').send_keys('022 2403 6606')
+        self.driver.find_element_by_id('id_unlisted_organization').send_keys('volunteer-org')
+        self.driver.find_element_by_xpath('//form[1]').submit()
+
+        # verify successful registration
+        self.assertNotEqual(self.driver.find_elements_by_class_name('messages'),
+                None)
+        self.assertEqual(self.driver.find_element_by_class_name('messages').text,
+                'You have successfully registered!')
+
+        # Try to register volunteer with incorrect phone number for country
         self.driver.get(self.live_server_url + self.volunteer_registration_page)
 
         self.driver.find_element_by_id('id_username').send_keys('volunteer-username-1')
         self.driver.find_element_by_id('id_password').send_keys('volunteer-password!@#$%^&*()_')
-        self.driver.find_element_by_id('id_first_name').send_keys('first-name')
-        self.driver.find_element_by_id('id_last_name').send_keys('last-name')
+        self.driver.find_element_by_id('id_first_name').send_keys('volunteer-first-name')
+        self.driver.find_element_by_id('id_last_name').send_keys('volunteer-last-name')
         self.driver.find_element_by_id('id_email').send_keys('volunteer-email1@systers.org')
         self.driver.find_element_by_id('id_address').send_keys('volunteer-address')
-        self.driver.find_element_by_id('id_city').send_keys('!@#$%^&*()_+city')
+        self.driver.find_element_by_id('id_city').send_keys('volunteer-city')
         self.driver.find_element_by_id('id_state').send_keys('volunteer-state')
-        self.driver.find_element_by_id('id_country').send_keys('volunteer-country')
-        self.driver.find_element_by_id('id_phone_number').send_keys('9999999999')
+        self.driver.find_element_by_id('id_country').send_keys('India')
+        self.driver.find_element_by_id('id_phone_number').send_keys('237937913')
```

Comment:
Yes, the user shouldn't be allowed to register in this case. This is the test scenario where user tries to register with a phone number not valid for the country chosen.